### ISSUE TITLE
make sure autobrief is turned on

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/NCEPLIBS-bufr#69]
We want the first sentance of every documentation block treated as the "brief" description, which has special value in doxygen.

**Comment by edwardhartnett** [comment]
OK, this is already on in the docs.

I build the docs, and look at ireadmg.f (which has a brief tag).

Then I take out the brief tag, and rebuild the docs. The HTML is the same, the first sentence is correctly identified as the brief description without the brief tag.

**Comment by jbathegit** [comment]
You're talking about JAVADOC_AUTOBRIEF=YES in the Doxyfile.in, correct?  Then shouldn't these brief descriptions also show up automatically in the "Files" tab in the HTML (i.e. in files.html)?  In other words, where it says "Here is a list of all documented files with brief descriptions:" shouldn't the brief descriptions all show up there as well, including for all of the subprograms which don't currently contain a `@brief` tag in the documentation?

This is a bit puzzling to me, because previously the `@brief` descriptions were showing up there for the 5 subprograms where we had that tag explicitly set.  But now I've been testing some other changes to the documentation, including to define user_guide.md as the mainpage within the USE_MDFILE_AS_MAINPAGE setting, rather than by using the `@mainpage` tag in the user_guide.md file itself (b/c if you have it in both places, Doxygen generates a conflict warning and neither ends up getting displayed).  The advantage to setting it via USE_MDFILE_AS_MAINPAGE is that it no longer generates a superfluous "user_guide" link in the HTML which points to nowhere.  But the disadvantage is that it now seems to have removed all of those aforementioned brief descriptions within the "Files" tab in the HTML, and I can't for the life of me figure out why that mainpage change would have any effect on the brief descriptions in the "Files" tab?

So that's a weird wrinkle, but in any case between JAVADOC_AUTOBRIEF=YES, BRIEF_MEMBER_DESC = YES, REPEAT_BRIEF = YES and ALWAYS_DETAILED_SEC = YES (i.e. all of these are currently set to YES), I would think that would be all we'd need(?)  

In summary, I think it's very useful to readers to have those brief descriptions show up under the "Files" tab, and I'd like to figure out what the issue is here, so any help would be greatly appreciated.  I've already looked extensively through online documentation for Doxygen, but I can't figure out what I'm doing wrong here.

**Comment by jbathegit** [comment]
I'm going to try something else, since we've now carefully separated the `@file` part of each docblock from the subprogram using a blank line, and since in this library pretty-much every file only contains a single subprogram.  In short, I'm going to try keeping the `@brief` tag, but move it up into the `@file` section and see if that makes it work...

**Comment by jbathegit** [comment]
Yep, that seems to have done the trick!  So it was a simple matter of file vs. subprogram defined within the file, which was all one big docblock previously without that blank line separating the file section from the subprogram section in the docblock.  All good now, and sorry for any confusion!  I'll now confirm that this all still works without the actual `@brief` tags in each section.

**Comment by edwardhartnett** [comment]
I have been playing with this too. It looks like we need the brief tag in the file section, but not in the subroutine section. Weird.

I will put up a PR with an example of what I mean...
